src/main/java/net/sf/jabref/importer/fileformat/MedlinePlainImporter.java

```diff
@@ -127,91 +126,93 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
                     }
                 }
                 String entry = current.toString();
+                if (entry.contains("-")) {
```

Comment:
I think the while-loop above this is already trying to check the current line for validity and tries to skip all the invalid ones. I would propose to extract a new method called `private boolean checkLineValidity(String line)` and do all the checks in there such as: empty line, line doesn't contain "-", "-" symbol not at 4th position, etc...
then the while loop above can also be replaced and you don't need to modify the counter variable "j". You can then simply skip any invalid lines with:
```
if (!checkLineValidity(lines[j])) {
    continue;
}
```